src/core/lib/request.js

```diff
@@ -83,20 +85,45 @@ function fetchRetry(url, headers, fetchOptions, resolve, reject, retryDelay = 30
             }
             wait(msDelay)
                 .then(() => {
-                    return fetchRetry(url, headers, fetchOptions, resolve, reject, retryDelay, retryLimit)
+                    return fetchRetry(stack, queryParams, fetchOptions, resolve, reject, retryDelay, retryLimit)
                 })
                 .catch(() => {
-                    return fetchRetry(url, headers, fetchOptions, resolve, reject, retryDelay, retryLimit)
+                    return fetchRetry(stack, queryParams, fetchOptions, resolve, reject, retryDelay, retryLimit)
                 })
         }
     }
+
     if (fetchOptions.debug)  fetchOptions.logHandler('info', { url: url, option: option});
-    fetch(url, option)
-        .then(function(response) {    
+
+    let request = {url, option};
+
+    let plugins = stack.plugins;
+    if (plugins && plugins != undefined) {
+
+        for (let index = 0; index < plugins.length; index++) {
+
+            if (typeof plugins[index].onRequest == 'function') {
+                request = plugins[index].onRequest(stack, request)
+            }
+        }
+    }
+
+    
+    fetch(request.url, request.option)
+        .then( function(response) {
+            
             if (fetchOptions.debug)  fetchOptions.logHandler('info', response);
-            let data = response.json();      
+            let data = response.json();
+
             if (response.ok && response.status === 200) {
```

Comment:
this entire block of code including the older one, looks pretty complex wiht too many if else, can we simplify it and make more readable?